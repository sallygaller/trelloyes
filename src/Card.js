// When the "delete card" button is clicked on any card, remove all references to that card from state.
// When the "add random card" button is clicked, generate a random card and add it to the state and the appropriate list.
// In the resources below, we've supplied you with a function for generating a random card.

import React from 'react';
import './Card.css';

export default function Card(props) {
    return (
        <div className='Card'>
            <h3>{props.title}</h3>
            <p>{props.content}</p>
            <button
                onClick={ () => props.onDeleteCard(props.id)}
                type='button'>
                Delete card
            </button>
        </div>
    )
}

Card.propTypes = {
    onClickDelete: () => {}
}