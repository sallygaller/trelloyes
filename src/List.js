// When the "delete card" button is clicked on any card, remove all references to that card from state.
// When the "add random card" button is clicked, generate a random card and add it to the state and the appropriate list.
// In the resources below, we've supplied you with a function for generating a random card.
import React from 'react';
import Card from './Card'
import './List.css';

export default function List(props) {
    const cards = Array.from(props.cards);
    console.log(cards)
    return (
        <section className='List'>
            <header>
                <h2>{props.header}</h2>
                <button
                    onClick={ () => props.onAddCard(props.id)}
                    type='button'>
                    Add random card
                </button>
            </header>
            <div className='List-cards'>
                {cards.map((card =>
                    <Card
                        //props for Cards component
                        key={card.id}
                        id={card.id}
                        title={card.title}
                        content={card.content}
                        onDeleteCard={props.onDeleteCard}
                    />
                ))}
            </div>
        </section>
    )
}

List.defaultProps = {
    onClickAdd: () => {}
}