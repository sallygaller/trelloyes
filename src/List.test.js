import renderer from 'react-test-renderer';
import React from 'react';
import ReactDOM from 'react-dom';
import List from './List';
import Card from './Card';

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<List />, div);
    ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected if the header is blank', () => {
    const tree = renderer
        .create(<List name="Header" header={""} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders the UI as expected if there are many cards', () => {
    const tree = renderer
        .create(<List name="List" cards={25} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});