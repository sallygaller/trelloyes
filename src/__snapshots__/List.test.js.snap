// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected if there are many cards 1`] = `
<section
  className="List"
>
  <header>
    <h2 />
  </header>
  <div
    className="List-cards"
  />
</section>
`;
